queue_test.go

```diff
@@ -0,0 +1,307 @@
+	if len(errs) != 0 {
+		assert.Fail(t, "errors are not empty")
+	}
+	assert.Equal(t, actualMsgs["5a8f3f37-3098-48f7-811a-f69d12f2b1be"].Headers["X-Request-Id"], "tid_test")
+	assert.NotEmpty(t, actualMsgs["5a8f3f37-3098-48f7-811a-f69d12f2b1be"].Headers["Message-Id"])
+	assert.Equal(t, actualMsgs["5a8f3f37-3098-48f7-811a-f69d12f2b1be"].Headers["Message-Type"], "cms-content-published")
+	assert.Equal(t, actualMsgs["5a8f3f37-3098-48f7-811a-f69d12f2b1be"].Headers["Content-Type"], "application/json")
+	assert.Equal(t, actualMsgs["5a8f3f37-3098-48f7-811a-f69d12f2b1be"].Headers["Origin-System-Id"], methodeSystemOrigin)
+	assert.Equal(t, actualMsgs["5a8f3f37-3098-48f7-811a-f69d12f2b1be"].Body, `{"contentUri":"http://methode-article-image-set-mapper.svc.ft.com/image-set/model/5a8f3f37-3098-48f7-811a-f69d12f2b1be","payload":{"uuid":"5a8f3f37-3098-48f7-811a-f69d12f2b1be","identifiers":[{"authority":"http://api.ft.com/system/FTCOM-METHODE","identifierValue":"5a8f3f37-3098-48f7-811a-f69d12f2b1be"}],"members":[{"uuid":"8ff1c7f4-a80b-4b8d-8821-b07ff1bfdf87"},{"uuid":"3bea853a-89b8-4831-80b3-8384e962f5dc"},{"uuid":"c6eeea75-748e-4b1c-a046-6e4c9d81ff25"}],"publishReference":"","lastModified":"","publishedDate":"2017-05-18T02:24:25Z","firstPublishedDate":"2017-05-18T02:24:00Z","canBeDistributed":"verify"},"lastModified":"2017-05-15T15:54:32.166Z"}`)
+
+	assert.Equal(t, actualMsgs["270c0151-7742-4c1e-b77e-a5557881a042"].Headers["X-Request-Id"], "tid_test")
+	assert.NotEmpty(t, actualMsgs["270c0151-7742-4c1e-b77e-a5557881a042"].Headers["Message-Id"])
+	assert.Equal(t, actualMsgs["270c0151-7742-4c1e-b77e-a5557881a042"].Headers["Message-Type"], "cms-content-published")
+	assert.Equal(t, actualMsgs["270c0151-7742-4c1e-b77e-a5557881a042"].Headers["Content-Type"], "application/json")
+	assert.Equal(t, actualMsgs["270c0151-7742-4c1e-b77e-a5557881a042"].Headers["Origin-System-Id"], methodeSystemOrigin)
+	assert.Equal(t, actualMsgs["270c0151-7742-4c1e-b77e-a5557881a042"].Body, `{"contentUri":"http://methode-article-image-set-mapper.svc.ft.com/image-set/model/270c0151-7742-4c1e-b77e-a5557881a042","payload":{"uuid":"270c0151-7742-4c1e-b77e-a5557881a042","identifiers":[{"authority":"http://api.ft.com/system/FTCOM-METHODE","identifierValue":"270c0151-7742-4c1e-b77e-a5557881a042"}],"members":[{"uuid":"667ee7f3-4f58-4080-a6f9-9b16b633dea8"},{"uuid":"a0513a50-08d1-43f6-af2b-7e7dc4d40b31"},{"uuid":"47e5a693-cd39-4ede-a016-244e6413a7fa"}],"publishReference":"","lastModified":"","publishedDate":"2017-05-18T02:24:25Z","firstPublishedDate":"2017-05-18T02:24:00Z","canBeDistributed":"verify"},"lastModified":"2017-05-15T15:54:32.166Z"}`)
+
+}
+
+type mockProducer struct{
+	mock.Mock
+}
+
+func (p *mockProducer) SendMessage(key string, msg producer.Message) error {
+	args := p.Called(key, msg)
+	return args.Error(0)
+}
+
+func (p *mockProducer) ConnectivityCheck() (string, error) {
+	args := p.Called()
+	return args.String(0), args.Error(1)
+}
+
+type mockMessageToNativeMapper struct {
```

Comment:
I see that this mock is also used quite a lot in the http mapper tests. Could we move all the mocks to a separate file? It feels confusing to me to have tests from a given file rely on mocks defined in the test file for another feature.